@here/harp-mapview/lib/MapView.ts

```diff
@@ -2721,29 +2694,15 @@ export class MapView extends THREE.EventDispatcher {
             !this.m_initialTextPlacementDone &&
             !this.m_firstFrameComplete &&
             !this.isDynamicFrame &&
-            !this.m_initialTextPlacementDone &&
             !this.m_themeIsLoading &&
             this.m_poiTableManager.finishedLoading &&
             this.m_visibleTiles.allVisibleTilesLoaded &&
             this.m_connectedDataSources.size + this.m_failedDataSources.size ===
-                this.m_tileDataSources.length
+                this.m_tileDataSources.length &&
+            !this.m_textElementsRenderer.initializing &&
+            !this.m_textElementsRenderer.loading
         ) {
-            const textElementRendererFinished =
-                this.m_textElementsRenderer !== undefined &&
-                this.m_textElementsRenderer.ready &&
-                !this.m_textElementsRenderer.loading;
-
-            if (!this.textElementsRendererRequested || textElementRendererFinished) {
-                if (textElementRendererFinished) {
-                    // Force label placement in this frame. This may change the loading state of the
-                    // TextElementsRenderer and the animation state of MapView.
-                    this.m_textElementsRenderer!.invalidateCache();
```

Comment:
I've removed the explicit cache invalidation from here, it shouldn't be needed. TextElementsRenderer takes care by itself to figure out when its text element cache must be updated. Therefore, it's enough to check that it's not initializing nor loading, to know that in this frame it'll do the final text placement (as long as tile loading is finished, which is checked as well).